x-pack/plugins/security_solution_serverless/server/cloud_security/defend_for_containers_metering.ts

```diff
@@ -1,148 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License
- * 2.0; you may not use this file except in compliance with the Elastic License
- * 2.0.
- */
-
-import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
-import type {
-  AggregationsAggregate,
-  SearchResponse,
-  SortResults,
-} from '@elastic/elasticsearch/lib/api/types';
-import type { Tier, UsageRecord } from '../types';
-import type { CloudSecurityMeteringCallbackInput } from './types';
-import { CLOUD_DEFEND, CLOUD_SECURITY_TASK_TYPE, CLOUD_DEFEND_HEARTBEAT_INDEX } from './constants';
-
-const BATCH_SIZE = 1000;
-const SAMPLE_WEIGHT_SECONDS = 3600; // 1 Hour
-
-export interface CloudDefendHeartbeat {
-  '@timestamp': string;
-  'agent.id': string;
-  event: {
-    ingested: string;
-  };
-}
-
-const buildMeteringRecord = (
-  agentId: string,
-  timestampStr: string,
-  taskId: string,
-  tier: Tier,
-  projectId: string
-): UsageRecord => {
-  const timestamp = new Date(timestampStr);
-  timestamp.setMinutes(0);
-  timestamp.setSeconds(0);
-  timestamp.setMilliseconds(0);
-  const creationTimestamp = new Date();
-  const usageRecord = {
-    id: `${projectId}_${agentId}_${timestamp.toISOString()}`,
-    usage_timestamp: timestampStr,
-    creation_timestamp: creationTimestamp.toISOString(),
-    usage: {
-      type: CLOUD_SECURITY_TASK_TYPE,
-      sub_type: CLOUD_DEFEND,
-      period_seconds: SAMPLE_WEIGHT_SECONDS,
-      quantity: 1,
-    },
-    source: {
-      id: taskId,
-      instance_group_id: projectId,
-      metadata: {
-        tier,
-      },
-    },
-  };
-
-  return usageRecord;
-};
-export const getUsageRecords = async (
-  esClient: ElasticsearchClient,
-  searchFrom: Date,
-  searchAfter?: SortResults
-): Promise<SearchResponse<CloudDefendHeartbeat, Record<string, AggregationsAggregate>>> => {
-  return esClient.search<CloudDefendHeartbeat>(
-    {
-      index: CLOUD_DEFEND_HEARTBEAT_INDEX,
```

Comment:
Is CLOUD_DEFEND_HEARTBEAT_INDEX still being used for anything else, should this index be cleaned up/deleted?

Reply:
as far as I know it deprecated only from serverless, @nick-alayil pleas keep me honest here.

Reply:
The plan is deprecate Defend for Containers from serverless and ECH 9.0 stack but it'll remain available in the ECH 8.18.